util/reggen/reg_block.py

```diff
@@ -245,6 +245,12 @@ def add_window(self, window: Window) -> None:
         assert self.offset <= window.offset
         self.offset = window.next_offset(self._addrsep)
 
+    def get_data_intg_passthru(self) -> bool:
+        # returns True if ANY window passes through data integrity
+        passthrus = [window.data_intg_passthru for window in self.windows]
+
+        return any(passthrus)
```

Comment:
Can you put this logic in `add_window`, updating a "has_passthrus" boolean field? That should avoid having to search through the list, which is maybe a bit simpler.

Reply:
good idea. let me do that.  Does this PR make sense to you otherwise?

Reply:
addressed comment.

Reply:
Yep, it looks good!